Assignment 4

Suruchibharti opened the pull request:
Create Dynamic Inventory Management System.c
- Added  structure for Dynamic Inventory Management System: 
1: Created Product structure to store ID, name, price, and quantity.
2: Added dynamic memory allocation using calloc().
3: Implemented input validation using fgets() and sscanf() to prevent invalid inputs.
4: Basic menu-driven interface added (Add, View,SearchById , SearchByName, SearchByPriceRange , UpdateById,  DeleteById and Exit).

Suruchibharti committed:
Create Dynamic Inventory Management System.c

Initial commit: Added  structure for Dynamic Inventory Management System: 
1: Created Product structure to store ID, name, price, and quantity.
2: Added dynamic memory allocation using calloc().
3: Implemented input validation using fgets() and sscanf() to prevent invalid inputs.
4: Basic menu-driven interface added (Add, View,SearchById , SearchByName, SearchByPriceRange , UpdateById,  DeleteById and Exit).

Suruchibharti committed:
Update Dynamic Inventory Management System.c

Suruchibharti committed:
Update Dynamic Inventory Management System.c

fixes the code for negiative product-quantity

Suryanshsrivastv commented:
Please check the input validation , Enter product name to search: Product name cannot be empty , also if the menu prints after each operation it would be easy to navigate

Suruchibharti committed:
Update Dynamic Inventory Management System.c

added printf after successfully addition of  a product in inventory

Suruchibharti commented:
> Please check the input validation , Enter product name to search: Product name cannot be empty , also if the menu prints after each operation it would be easy to navigate

@Suryanshsrivastv , I’ve fixed the input validation for the “Enter product name to search” part — now if the user enters an empty name, they’ll get another chance to type it correctly.
Also, the menu already reappears automatically after each operation since it’s inside the do-while loop, so navigation is smoother now. So, kindly review it again.

Suruchibharti committed:
Update Dynamic Inventory Management System.c

 fixed the input validation for the “Enter product name to search” part — now if the user enters an empty name, they’ll get another chance to type it correctly.

Suruchibharti committed:
Update Dynamic Inventory Management System.c

Menu reappears after every operation for smooth functioning of program.

Suruchibharti committed:
Update Dynamic Inventory Management System.c

Guptaji2003 commented on Assignment-4/Dynamic Inventory Management System.c:
```diff
@@ -0,0 +1,461 @@
+    printf("Enter maximum price: ");
+    scanf("%f", &maxPrice);
+    clearInputBuffer();
+
```
I think you should use loop that run until valid input.

Suruchibharti replied:
> I think you should use loop that run until valid input.

Sure, I have implemented this in my code . Kindly review it .

Suruchibharti committed:
Update Dynamic Inventory Management System.c

added input validation for empty product name

Suruchibharti committed:
Update Dynamic Inventory Management System.c

prventing unitialized reading in checkIdExist funciton
